src/components/board/lcd/index.tsx

```diff
@@ -0,0 +1,116 @@
+import Board from 'board'
+import React from 'react'
+import './style.css'
+
+type LcdState = {
+  displayValueRow0: String[]
+  displayValueRow1: String[]
+  greenOpacity: number
+  redOpacity: number
+  blueOpacity: number
+}
+
+export class LcdComponent extends React.Component<{}, LcdState> {
+  private static readonly LCD_POSITION_COUNT = 40
+
+  constructor(props: {}) {
+    super(props)
+    Board.processor.on('afterCycle', () => this.update())
+    this.state = this.getState()
+  }
+
+  private update() {
+    return this.setState(this.getState())
+  }
+
+  private getState() {
+    let allValuesRow0: String[] = []
+    for (
+      let i = LcdComponent.LCD_POSITION_COUNT - 1;
+      i >= LcdComponent.LCD_POSITION_COUNT / 2;
+      i--
+    ) {
+      allValuesRow0.push(Board.lcdDisplay.getDisplayValue(i))
+    }
+    let allValuesRow1: String[] = []
+    for (let i = LcdComponent.LCD_POSITION_COUNT / 2 - 1; i >= 0; i--) {
+      allValuesRow1.push(Board.lcdDisplay.getDisplayValue(i))
+    }
+    const brightness = Board.lcdDisplay.getColour()
+    let redBright = this.byteValueToPercent(
+      this.scaleDownToByteValue(brightness[0].value)
+    )
+    let greenBright = this.byteValueToPercent(
+      this.scaleDownToByteValue(brightness[1].value)
+    )
+    let blueBright = this.byteValueToPercent(
+      this.scaleDownToByteValue(brightness[2].value)
+    )
+    let sumAllBright = greenBright + redBright + blueBright
+
+    if (sumAllBright != 0 && sumAllBright > 1.0) {
+      redBright = redBright / sumAllBright
```

Comment:
I think the if-condition could be shortened to 'sumAllBright > 1.0' since this includes the first condition.

Reply:
you are right!